src/ensembl/src/content/app/browser/track-panel/trackPanelActions.ts

```diff
@@ -3,9 +3,42 @@ import { ThunkAction } from 'redux-thunk';
 import { Action, ActionCreator } from 'redux';
 
 import { RootState } from 'src/store';
-import { TrackSet } from './trackPanelConfig';
+import { TrackSet, GenomeTrackStates } from './trackPanelConfig';
+import trackPanelStorageService from './track-panel-storage-service';
 import browserStorageService from '../browser-storage-service';
-import { getBrowserActiveGenomeId } from '../browserSelectors';
+import {
+  getBrowserActiveGenomeId,
+  getBrowserActiveEnsObject,
+  getBrowserTrackStates
+} from '../browserSelectors';
+import { getActiveGenomePreviouslyViewedObjects } from './trackPanelSelectors';
+
+import { Bookmark } from './trackPanelState';
+import { EnsObject } from 'src/ens-object/ensObjectTypes';
+import { getFormattedLocation } from 'src/shared/helpers/regionFormatter';
+
+const buildBookmark = (
+  ensObject: EnsObject,
+  trackStates: GenomeTrackStates
+): Bookmark => {
+  let bookmarkLabel = ensObject.object_type === 'gene' ? ensObject.label : '';
+
+  if (!bookmarkLabel && ensObject.stable_id) {
+    bookmarkLabel = ensObject.label;
+  } else if (!bookmarkLabel) {
+    bookmarkLabel = getFormattedLocation(ensObject.location);
+  }
```

Comment:
Currently all ens objects, including regions, that we get from the backend have a `label` field. There is no need for all this logic.

Reply:
Removed it. Thanks